test/e2e-go/upgrades/application_support_test.go

```diff
@@ -448,18 +448,24 @@ int 1
 	a.NoError(err)
 	signedTxn, err = client.SignTransactionWithWallet(wh, nil, tx)
 	a.NoError(err)
-	_, err = client.BroadcastTransaction(signedTxn)
-	a.NoError(err)
-
-	curStatus, err = client.Status()
+	txid, err := client.BroadcastTransaction(signedTxn)
 	a.NoError(err)
-
-	round = curStatus.LastRound
-
-	client.WaitForRound(round + 2)
-	pendingTx, err = client.GetPendingTransactions(1)
-	a.NoError(err)
-	a.Equal(uint64(0), pendingTx.TotalTransactions)
+	for {
```

Comment:
It would still be nice to have some round/time limit that would cause the test to fail so that it always terminates.

I'd suggest something like 10 rounds, which is large enough that the txn should always be confirmed before then if everything's working properly.